src/main/java/com/google/sps/NodeMutationFilterTest.java

```diff
@@ -0,0 +1,173 @@
+  /** Basic test for including mutliple relevant nodes for getMutationIndicesOfNode */
+  @Test
+  public void getMutationsOfBasic() {
+    Mutation addAB =
+        Mutation.newBuilder()
+            .setType(Mutation.Type.ADD_EDGE)
+            .setStartNode("A")
+            .setEndNode("B")
+            .build();
+    Mutation removeAB =
+        Mutation.newBuilder()
+            .setType(Mutation.Type.DELETE_EDGE)
+            .setStartNode("A")
+            .setEndNode("B")
+            .build();
+    Mutation removeC =
+        Mutation.newBuilder().setType(Mutation.Type.DELETE_NODE).setStartNode("C").build();
+    List<Mutation> mutList = new ArrayList<>();
+    mutList.add(addAB);
+    mutList.add(removeAB);
+    mutList.add(removeC);
+    List<Integer> truncatedList = Utility.getMutationIndicesOfNode("A", mutList);
+
+    Assert.assertEquals(2, truncatedList.size());
+    Assert.assertTrue(truncatedList.contains(0));
+    Assert.assertTrue(truncatedList.contains(1));
+    Assert.assertFalse(truncatedList.contains(2));
+  }
+
+  /** Test that a null query returns an empty list */
+  @Test
+  public void getMutationsOfNull() {
+    Mutation addAB =
+        Mutation.newBuilder()
+            .setType(Mutation.Type.ADD_EDGE)
+            .setStartNode("A")
+            .setEndNode("B")
+            .build();
+    List<Mutation> mutList = new ArrayList<>();
+    mutList.add(addAB);
+
+    List<Integer> truncatedList = Utility.getMutationIndicesOfNode(null, mutList);
+
+    Assert.assertNotNull(truncatedList); // Should not be null
+    Assert.assertEquals(0, truncatedList.size());
+    Assert.assertFalse(truncatedList.contains(0));
+  }
+
+  // The following test cases are for the getNextGreatestNumIndex function in
+  // Utility.
+  // Each will be on an list of even length and a list of odd length.
+  // The comment will apply to the two immediate test cases under it!
+
+  /** If no elements are greater, you get -1. */
+  @Test
+  public void testNextGreatestNumNoneExistEven() {
+    int ans = Utility.getNextGreatestNumIndex(lst1, 20);
+    Assert.assertEquals(-1, ans);
+  }
+
+  @Test
+  public void testNextGreatestNumNoneExistOdd() {
+    int ans = Utility.getNextGreatestNumIndex(lst1, 20);
+    Assert.assertEquals(-1, ans);
+  }
+
+  /** Ensures function gets the index rather than the actual value */
+  @Test
+  public void testIndexNotValueEven() {
+    int ans = Utility.getNextGreatestNumIndex(lst1, 8);
+    Assert.assertEquals(3, ans);
+  }
+
+  @Test
+  public void testIndexNotValueOdd() {
+    int ans = Utility.getNextGreatestNumIndex(lst2, 8);
+    Assert.assertEquals(2, ans);
+  }
+
+  /**
+   * When the tgt value does exist in the middle, value found must get strictly greater (value's
+   * index is returned)
+   */
+  @Test
+  public void testInnerExistsEven() {
+    int ans = Utility.getNextGreatestNumIndex(lst1, 4);
+    Assert.assertEquals(2, ans);
+  }
+
+  @Test
+  public void testInnerExistsOdd() {
```

Comment:
This test would make more sense if you "controlled" what you were searching for (like searching for 4 in both cases).